src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -5397,6 +5397,34 @@ public class MyType
         Assert.Collection(result.Diagnostics, d => { Assert.Equal("CS0411", d.Id); });
     }
 
+    [Fact, WorkItem("https://github.com/dotnet/aspnetcore/issues/48526")]
+    public void EventCallbackOfT_Array()
+    {
+        // Arrange
+        AdditionalSyntaxTrees.Add(Parse("""
+            using Microsoft.AspNetCore.Components.Forms;
+
+            namespace Test;
+
+            public class MyComponent<TItem> : InputBase<TItem[]>
+            {
+                protected override bool TryParseValueFromString(string value, out TItem[] result, out string validationErrorMessage) => throw null;
+            }
+            """));
+
+        // Act
+        var generated = CompileToCSharp("""
+            <MyComponent @bind-Value="Selected" />
+
+            @code {
+                string[] Selected { get; set; } = Array.Empty<string>();
+            }
+            """);
+
+        // Assert
+        CompileToAssembly(generated);
```

Comment:
I think it would help me better understand what's going on here if we asserted baselines on this code.

Reply:
The rarely-seen useful force push :D. Thanks.